cmd/agent/main.go

```diff
@@ -86,6 +86,8 @@ func main() {
 	}
 
 	if cfg.Integrations.Enabled {
+		// If a client auth is defined in TLS assume HTTPS/TLS
+		cfg.Integrations.SeverUsingTLS = cfg.Server.HTTPTLSConfig.ClientAuth != ""
```

Comment:
Can you move the logic here to ApplyDefaults in `pkg/config/config.go`?

Reply:
Done!